decidim-core/lib/decidim/participatory_space_resourceable.rb

```diff
@@ -47,7 +47,7 @@ def linked_participatory_space_resources(resource_name, link_name)
              .joins(:participatory_space_resource_links_to)
              .where(decidim_participatory_space_links: { name: link_name, from_id: id, from_type: self.class.name })
 
-        klass.where(id: from).or(klass.where(id: to))
+        klass.where(id: from).or(klass.where(id: to)).order(:weight)
```

Comment:
How about something like:
```ruby
if klass.column_names.include?("weight")
  query.order(:weight)
else
  query.order(created_at: :desc)
end
```

Also, please add a regression test for this.

Reply:
something similar i did here ... 
https://github.com/decidim/decidim/pull/10062/commits/0d6388730bd62021a875f5e74e26578cbcb808bd
that was one of the original fix proposed ... 

I will think on how the regressions should look like.